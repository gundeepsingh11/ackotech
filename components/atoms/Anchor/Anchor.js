// @flow
/**
 * Anchor component to handle all the click events
 * @module Anchor
 */
import Link from 'next/link';
import type from 'react';
import withStyles from '../../../lib/withStyles';
import styles from './Anchor.style';

const Anchor = props => {
  const {
    children,
    to,
    as,
    className,
    noLink,
    inheritedStyles,
    handleLinkClick,
    shallow,
    scroll,
    forceClick,
    hoverAnimation,
    viewmorebutton,
    target,
    ...other
  } = props;

  return (
    <Link href={to} as={as} shallow={shallow} scroll={scroll}>
      {forceClick ? (
        <a
          className={`${className} ${hoverAnimation ? 'hover-animation' : ''}`}
          onClick={handleLinkClick}
          target={target}
          rel={target ? 'noopener' : null}
          {...other}>
          {children}
        </a>
      ) : (
        <a
          className={`${className} ${hoverAnimation ? 'hover-animation' : ''}`}
          {...other}>
          {children}
        </a>
      )}
    </Link>
  );
};

Anchor.defaultProps = {
  noLink: false,
  to: '',
  as: '',
  inheritedStyles: '',
  shallow: false,
  scroll: true,
  forceClick: false,
  handleLinkClick: 'function',
  viewmorebutton: 'true',
};

export default withStyles(Anchor, styles);

export { Anchor as AnchorVanilla };
