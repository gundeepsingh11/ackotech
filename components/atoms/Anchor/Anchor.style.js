import { css } from 'styled-components';

export default css`
  cursor: pointer;
  text-decoration: none;
  color: ${props => props.theme.textColorWhite};
  padding: ${props => (props.isButtonLink ? '0.75rem 1.5rem' : '0')};
  display: block;

  .icon-next-arrow-white {
    font-size: ${props => props.theme.fontExtraSmall};

    &::before {
      color: ${props => props.theme.textColorPrimary};
    }
  }

  .animate-right {
    margin-right: 9px;
    animation: translate-left-in 0.3s 1 cubic-bezier(0.7, 0, 0.3, 1);
    position: relative;

    + span {
      margin-left: 0.5rem;
    }

    + p {
      margin-left: 0.5rem;
    }

    + div {
      margin-left: 0.5rem;
    }
  }

  &:hover {
    .animate-right {
      animation: translate-left-out 0.3s 1 cubic-bezier(0.7, 0, 0.3, 1);
      left: 0.6875rem;
    }
  }

  /* Keyframes */
  @keyframes translate-left-in {
    0% {
      left: 9px;
    }

    100% {
      left: 0;
    }
  }

  @keyframes translate-left-out {
    0% {
      left: 0;
    }

    100% {
      left: 9px;
    }
  }

  ${props =>
    props.secondary
      ? `
        font-weight: ${props.theme.fontSemiBold}
        color: ${props.theme.linkColorSecondary}
      `
      : ''};
  ${props =>
    props.readMoreButton
      ? `
        color: ${props.theme.textColorPrimary};
          margin-left: 10px;
      `
      : ''};
  ${props =>
    props.viewmorebutton
      ? `
        color: ${props.theme.textColorPrimary};
        margin-right: 15px;
      `
      : ''};

  &.generic-page-anchor {
    .icon-right-arrow-large {
      &::before {
        color: ${props => props.theme.textColorPrimary};
      }
    }
  }
`;
