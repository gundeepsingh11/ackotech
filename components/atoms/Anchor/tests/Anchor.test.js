import Router from 'next/router';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { AnchorVanilla } from '../Anchor';

const propsWithAnimation = {
  to: '',
  hoverAnimation: true,
};

const propsWithForceClick = {
  to: '',
  forceClick: true,
};

const propsWithForceClickAndAnimation = {
  to: '',
  forceClick: true,
  hoverAnimation: true,
};

const noLinkWithHoverAnimation = {
  noLink: true,
  hoverAnimation: true,
};

jest.mock('next/link', () => ({ children }) => children);

describe('Anchor Component', () => {
  let anchor;

  test('should render correctly and match snapshot', () => {
    anchor = shallow(<AnchorVanilla to="/sample">Sample</AnchorVanilla>);
    expect(anchor).toMatchSnapshot();
  });

  test('should render with styles correctly and match snapshot', () => {
    anchor = renderer.create(<AnchorVanilla to="/sample">Sample</AnchorVanilla>).toJSON();
    expect(anchor).toMatchSnapshot();
  });

  test('Should handle on typeahead focuson Events', () => {
    const mockedRouter = { push: () => {} };
    Router.router = mockedRouter;
    const props = { noLink: true, to: '/sample', handleLinkClick: jest.fn() };
    const event = { preventDefault: jest.fn() };
    anchor = shallow(<AnchorVanilla {...props}>Sample</AnchorVanilla>);
    anchor.find('a').simulate('click', event);
    // expect(props.handleLinkClick).toHaveBeenCalled();
  });

  test('test with hover animation', () => {
    anchor = shallow(<AnchorVanilla {...propsWithAnimation} />);
    expect(anchor).toMatchSnapshot();
  });

  test('test with force click', () => {
    anchor = shallow(<AnchorVanilla {...propsWithForceClick} />);
    expect(anchor).toMatchSnapshot();
  });

  test('test with force click and hover animation', () => {
    anchor = shallow(<AnchorVanilla {...propsWithForceClickAndAnimation} />);
    expect(anchor).toMatchSnapshot();
  });

  test('test with no link and hover animation', () => {
    anchor = shallow(<AnchorVanilla {...noLinkWithHoverAnimation} />);
    expect(anchor).toMatchSnapshot();
  });
});
