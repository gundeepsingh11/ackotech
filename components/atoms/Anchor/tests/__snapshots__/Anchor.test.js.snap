// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Component should render correctly and match snapshot 1`] = `
<Component
  as=""
  href="/sample"
  scroll={true}
  shallow={false}
>
  <a
    className="undefined "
  >
    Sample
  </a>
</Component>
`;

exports[`Anchor Component should render with styles correctly and match snapshot 1`] = `
<a
  className="undefined "
>
  Sample
</a>
`;

exports[`Anchor Component test with force click 1`] = `
<Component
  as=""
  href=""
  scroll={true}
  shallow={false}
>
  <a
    className="undefined "
    onClick="function"
    rel={null}
  />
</Component>
`;

exports[`Anchor Component test with force click and hover animation 1`] = `
<Component
  as=""
  href=""
  scroll={true}
  shallow={false}
>
  <a
    className="undefined hover-animation"
    onClick="function"
    rel={null}
  />
</Component>
`;

exports[`Anchor Component test with hover animation 1`] = `
<Component
  as=""
  href=""
  scroll={true}
  shallow={false}
>
  <a
    className="undefined hover-animation"
  />
</Component>
`;

exports[`Anchor Component test with no link and hover animation 1`] = `
<a
  className="undefined hover-animation"
  href=""
  rel={null}
/>
`;
